python/fledge/services/core/api/statistics.py

```diff
@@ -148,3 +150,83 @@ async def get_statistics_history(request):
+      """
+    params = request.query
+    if 'periods' not in params:
+        raise web.HTTPBadRequest(reason="periods request parameter is required")
+    if 'statistics' not in params:
+        raise web.HTTPBadRequest(reason="statistics request parameter is required")
+
+    if params['periods'] == '':
+        raise web.HTTPBadRequest(reason="periods cannot be an empty. Also comma separated list of values required "
+                                        "in case of multiple periods of time")
+    if params['statistics'] == '':
+        raise web.HTTPBadRequest(reason="statistics cannot be an empty. Also comma separated list of statistics values "
+                                        "required in case of multiple assets")
+
+    periods = params['periods']
+    period_split_list = list(filter(None, periods.split(',')))
+    if not all(p.isdigit() for p in period_split_list):
+        raise web.HTTPBadRequest(reason="periods should contain numbers")
+    # 1 week = 10080 mins
+    if any(int(p) > 10800 for p in period_split_list):
+        raise web.HTTPBadRequest(reason="The maximum allowed value for a period is 10080 minutes")
+
+    stats = params['statistics']
+    stat_split_list = list(filter(None, [x.upper() for x in stats.split(',')]))
+    storage_client = connect.get_storage_async()
+    # To find the interval in secs from stats collector schedule
+    scheduler_payload = PayloadBuilder().SELECT("schedule_interval").WHERE(
+        ['process_name', '=', 'stats collector']).payload()
+    result = await storage_client.query_tbl_with_payload('schedules', scheduler_payload)
+    if len(result['rows']) > 0:
+        scheduler = Scheduler()
+        interval_days, interval_dt = scheduler.extract_day_time_from_interval(result['rows'][0]['schedule_interval'])
+        interval = datetime.timedelta(days=interval_days, hours=interval_dt.hour, minutes=interval_dt.minute,
+                                      seconds=interval_dt.second)
+        interval_in_secs = interval.total_seconds()
+    else:
+        raise web.HTTPNotFound(reason="No stats collector schedule found")
+    ts = datetime.datetime.now().timestamp()
+    resp = []
+    for x, y in [(x, y) for x in period_split_list for y in stat_split_list]:
+        time_diff = ts - int(x)
+        dt = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(time_diff))
+        # FIXME:
+        # For example:
+        # time_diff = 1590066814.037321
+        # ERROR: PostgreSQL storage plugin raising error: ERROR:  invalid input syntax for type timestamp with time zone: "1590066814.037321"
+        # "where": {"column": "history_ts", "condition": ">=", "value": "1590066814.037321"} - Payload works with sqlite engine BUT not with postgres
+        # To overcome above problem on postgres - I have used "dt = 2020-05-21 13:13:34" - but I see some deviations in results for both engines when we use datetime format
```

Comment:
Also please review the addressed problem and let me know what's the actual fix for it

Reply:
Let's raise a separate Jirs for the Postgres differences. I suspect we have a backlog of things that need to be looked at in Postgres and should aim to handle them in a block.

Reply:
FOGL-4102 created